app/features/dm.ts

```diff
@@ -77,49 +77,44 @@ export async function sendDirectMessages(args: {
 
 export function getAllEncryptedMessagesOf(
     publicKey: PublicKey,
-    relay: ConnectionPool,
+    pool: ConnectionPool,
+    dmController: DirectedMessageController,
 ) {
-    const stream1 = getAllEncryptedMessagesSendBy(
-        publicKey,
-        relay,
-    );
-    const stream2 = getAllEncryptedMessagesReceivedBy(
-        publicKey,
-        relay,
-    );
-    return merge(stream1, stream2);
+    const lastestDMSendByMe = dmController.getLastestMessage("send_by_me");
+    const lastestDMReceivedByMe = dmController.getLastestMessage("received_by_me");
+    const since = (create_time?: Date) => {
+        if (create_time) {
+            return create_time.getTime() / 1000;
+        }
+        return hours_ago(15);
```

Comment:
When no since, do not pass in a since because it's more likely to be a newly logged in device. The implementation in main branch has this bug.

Reply:
oh, you are right